### PR TITLE
Rust Backend Serialization

### PR DESCRIPTION
**Pull request by ilitteri**

### CODE REVIEW

**Comment by ilitteri** [comment]
[Rust Backend Serialization](https://trello.com/c/2oIuoNTE/55-rust-backend-serialization)

File: src/gnark_backend_wrapper/groth16/serialize.rs [hunk 1]
```diff
@@ -1,73 +1,90 @@
...
-            ark_serialize::CanonicalSerialize::serialize_uncompressed(
-                coefficient,
-                &mut serialized_coefficient,
-            )
-            .map_err(|e| serde::ser::Error::custom(e.to_string()))?;
-            serializable_add_terms.push((hex::encode(serialized_coefficient), *sum));
-        }
+pub fn serialize_felt_unchecked(felt: &gnark_backend::groth16::Fr) -> Vec<u8> {
+    let mut serialized_felt = Vec::new();
+    #[allow(clippy::unwrap_used)]
```
Comment (ilitteri) [inline src/gnark_backend_wrapper/groth16/serialize.rs:49]: I'll make a ticket for this

File: src/gnark_backend_wrapper/groth16/serialize.rs [hunk 2]
```diff
@@ -1,73 +1,90 @@
...
-            ark_serialize::CanonicalSerialize::serialize_uncompressed(value, &mut serialized_value)
-                .map_err(|e| serde::ser::Error::custom(e.to_string()))?;
-            serializable_values.push(hex::encode(serialized_value));
-        }
-
-        s.serialize_field("gates", &self.gates)?;
-        s.serialize_field("public_inputs", &self.public_inputs)?;
-        s.serialize_field("values", &serializable_values)?;
-        s.serialize_field("num_variables", &self.num_variables)?;
+        let mut s = serializer.serialize_struct("MulTerm", 3)?;
```
Comment (matias-gonz) [inline src/gnark_backend_wrapper/groth16/serialize.rs:19]: What is the 3?

**Review by matias-gonz — COMMENTED** [review]
👍

File: src/gnark_backend_wrapper/groth16/serialize.rs (hunk 2 above)
Comment (ilitteri) [reply src/gnark_backend_wrapper/groth16/serialize.rs:19]: `3` is the number of fields in the struct to serialize (`MulTerm` in this case). I'll encapsulate the number in a variable.

**Review by jrchatruc — APPROVED** [review]

**Pull request merged**